doc/specs/#120,121,119-Update-Uninstall-List.md

```diff
@@ -0,0 +1,324 @@
+
+#### Enumerating installed applications
+
+##### MSI and EXE
+
+The Windows Package Manager will be enumerate the installed MSI and EXE files by searching the ProductCodes in the uninstall locations.
+The `ProductCode` will be read from the following locations:
+
+```plaintext
+64bit application:
+`HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\`
+32bit application:
+`HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\`
+```
+
+Using the `ProductCode`, the Windows Package Manager will be able to determine if there is an update available in the community repository.
+
+##### Windows Store applications and MSIX
+
+`Store apps` and `MSIX` installers, the Windows Package Manager will use the `PackageFamilyName` to determine which applications are installed. For each app installed, we will query the store to see if there is an update available.
+
+> Open Issue: What exactly can we display? Will we have the version? Need to close with Yao.
+
+#### Mapping Ids To YAML files
+
+Once the local files are enumerated, they must be mapped to the supported files in Community repository. To do this will will rely on the uninstall string used by the installer formats.For example:  `ProductCode` and `PackageFamilyName` as appropriate.
+
+* [ProductCode](https://docs.microsoft.com/windows/win32/msi/product-codes)
+* [PackageFamilyName](https://docs.microsoft.com/mem/configmgr/protect/deploy-use/find-a-pfn-for-per-app-vpn)
+* `ZIP` undefined
+* `Standalone exe` undefined
+
+> Open issue: Need a better enum for zip and standalone exes.
+
+For complete details on `ProductCode` and `PackageFamilyName` see [YAMLSPEC](..\ManifestSpecv0.2.md)
+
+#### Determining update availability
+
+Once the mapping is complete between the applications installed and the entry in the community repository, the client can determine if an update is available.
+
+| Sequence | Result |
+|---|---|
+| Installed version >= Community version | No update |
+| Installed version < Community version | Display newer version |
+| Community version is 'latest' | *Get latest |
+
+*If we can calculate the community version, to compare against the installed version, that is of course the preference.
+Proposal for those applications we cannot tell if there is an update. Compare hashes. If hashes match, then assume there is no update available. If hashes do not match, then display Get Latest.
+
+For Windows Store, the client will need to query the store to see if an update is available.
+
+##### Freshness of the database
+
+For `list` to display accurate information, the database of content must be up to date. The default experience for a downloaded catalog will be to refresh every 5 minutes. The default experience for RESTFUL apis will be to query the server at `list` execution.
+
+### Update
```

Comment:
I think winget should use keywork like apt of Debian.

- Update: for refresh the catalog
- Upgrade: for actual installation operation of new version

This way we could remove the refresh frequency as winget user does not expect winget run in background as a CLI tool usually run and exit